pkg/vsphere/tasks/waiter.go

```diff
@@ -89,6 +102,56 @@ func WaitForResult(ctx context.Context, f func(context.Context) (Task, error)) (
 	}
 }
 
+func fixTask(ctx context.Context, sess *session.Session, target object.Reference) error {
+	switch vmm := target.(type) {
+	case *object.VirtualMachine:
+		vm := vm.NewVirtualMachine(ctx, sess, target.Reference())
```

Comment:
We still rely on session to get VMFolder, if it's ESXi, or we do not use resource pool. Remove Session cannot simply make it easy.